packages/ripple-keypairs/src/index.ts

```diff
@@ -122,40 +141,79 @@ function deriveKeypair(
   const messageToVerify = hash('This test message should verify.')
   const signature = method.sign(messageToVerify, keypair.privateKey)
   /* istanbul ignore if */
-  if (method.verify(messageToVerify, signature, keypair.publicKey) !== true) {
+  if (!method.verify(messageToVerify, signature, keypair.publicKey)) {
     throw new Error('derived keypair did not generate verifiable signature')
   }
   return keypair
 }
 
-function getAlgorithmFromKey(key): 'ed25519' | 'ecdsa-secp256k1' {
+/**
+ * Determines the algorithm associated with a given key (public/private).
+ *
+ * | Curve     | Type        | Prefix | Length | Description                                           | Algorithm       |
+ * |-----------|-------------|:------:|:------:|-------------------------------------------------------|----------------:|
+ * | ed25519   | Private     |  0xED  |   33   | prefix + Uint256LE (0 < n < order )                   |         ed25519 |
+ * | ed25519   | Public      |  0xED  |   33   | prefix + 32 y-bytes                                   |         ed25519 |
+ * | secp256k1 | Public (1)  |  0x02  |   33   | prefix + 32 x-bytes                                   | ecdsa-secp256k1 |
+ * | secp256k1 | Public (2)  |  0x03  |   33   | prefix + 32 x-bytes (y is odd)                        | ecdsa-secp256k1 |
+ * | secp256k1 | Public (3)  |  0x04  |   65   | prefix + 32 x-bytes + 32 y-bytes                      | ecdsa-secp256k1 |
+ * | secp256k1 | Private (1) |  None  |   32   | Uint256BE (0 < n < order)                             | ecdsa-secp256k1 |
+ * | secp256k1 | Private (2) |  0x00  |   33   | prefix + Uint256BE (0 < n < order)                    | ecdsa-secp256k1 |
+ *
+ * Note: The 0x00 prefix for secpk256k1 Private (2) essentially 0 pads the number
+ *       and the interpreted number is the same as 32 bytes.
+ *
+ * @param key - hexadecimal string representation of the key.
+ * @returns {'ed25519' | 'ecdsa-secp256k1'} algorithm for signing/verifying
+ * @throws Error when key is invalid
+ */
+function getAlgorithmFromKey(key: HexString): 'ed25519' | 'ecdsa-secp256k1' {
```

Comment:
Should something be added to the `HISTORY` talking about how this handles errors better now?

Reply:
done